test/api/qcFlags.test.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { expect } = require('chai');
+const request = require('supertest');
+const { server } = require('../../lib/application');
+const { resetDatabaseContent } = require('../utilities/resetDatabaseContent.js');
+
+module.exports = () => {
+    before(resetDatabaseContent);
+    describe('GET /api/qcFlags/:id', () => {
+        it('should successfuly fetch one QC flag type', async () => {
+            const response = await request(server).get('/api/qcFlags/4');
+            expect(response.status).to.be.equal(200);
+            const { data: qcFlag } = response.body;
+            expect(qcFlag).to.be.eql({
+                id: 4,
+                from: (1647914400 + 10000) * 1000,
+                to: (1647914400 + 10000) * 1000,
+                comment: 'Some qc comment 4',
+                createdAt: 1707825439000,
+                updatedAt: 1707825439000,
+                runNumber: 1,
+                dplDetectorId: 1,
+                createdById: 2,
+                createdBy: { id: 2, externalId: 456, name: 'Jan Jansen' },
+                flagTypeId: 13,
+                flagType: { id: 13, name: 'Bad', method: 'Bad', bad: true, archived: false, color: null },
+            });
+        });
+
+        it('should send error that QC flag type with given id cannot be found', async () => {
+            const response = await request(server).get('/api/qcFlags/99999');
+            expect(response.status).to.be.equal(404);
+            const { errors } = response.body;
+            expect(errors).to.be.eql([
+                {
+                    status: 404,
+                    title: 'Not found',
+                    detail: 'Quality Control Flag with this id (99999) could not be found',
+                },
+            ]);
+        });
+    });
+
+    describe('GET /api/qcFlags', () => {
+        it('should successfuly fetch all data', async () => {
+            const response = await request(server).get('/api/qcFlags');
+            expect(response.status).to.be.equal(200);
+            const { data, meta } = response.body;
+            expect(meta).to.be.eql({ page: { totalCount: 5, pageCount: 1 } });
+            expect(data).to.be.an('array');
+            expect(data).to.be.lengthOf(5);
+            const oneFetchedFlag = data.find(({ id }) => id === 4);
```

Comment:
Please check the ids of all, no need to check all the properties I think

Reply:
applied, checking one full object left, there is not other check for whether the structure is correct